From 1ce32886bf83c45d6487fc90dc51f8715bf70982 Mon Sep 17 00:00:00 2001
From: Bogdan Hamciuc <bogdan.hamciuc@nxp.com>
Date: Thu, 7 Nov 2019 17:59:30 +0200
Subject: [PATCH 027/269] s32g: Update SRAM base address

Use 0x3400_0000 as the SRAM base addr.
Define the BL33 address as a function of the SRAM base addr.
Remove unused option which complicated these definitions.

Upstream-Status: Pending 

Signed-off-by: Bogdan Hamciuc <bogdan.hamciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
Issue: ALB-4277
---
 plat/s32g/include/platform_def.h | 14 ++++----------
 1 file changed, 4 insertions(+), 10 deletions(-)

diff --git a/plat/s32g/include/platform_def.h b/plat/s32g/include/platform_def.h
index 6b4513ca5..c2bcc9f93 100644
--- a/plat/s32g/include/platform_def.h
+++ b/plat/s32g/include/platform_def.h
@@ -79,7 +79,7 @@
 /* SRAM is actually at 0x3400_0000; we are just mirroring it in the
  * Virtual Code RAM
  */
-#define S32G_SRAM_BASE		0x38000000
+#define S32G_SRAM_BASE		0x34000000
 #define S32G_SRAM_SIZE		0x00A00000
 #define S32G_DDR0_BASE		0x80000000
 /* FIXME this should be a compile-time option; in addition, on S32G we actually
@@ -105,14 +105,6 @@
 /* BL31 and BL33 location in SRAM
  */
 
-/* U-boot address in SRAM */
-#define S32G_BL33_IMAGE_BASE		0x38020000
-
-#ifdef CONFIG_S32G_BL31_BEFORE_BL32	/* TODO add this to platform.mk */
-#define S32G_BL31_OFF_IN_SRAM		0x00002000
-/* Make sure bl31 does not overlap with u-boot */
-#define BL31_LIMIT			(S32G_BL33_IMAGE_BASE - 1)
-#else
 /* BL31 is located *after* BL32 in SRAM, where there is more space potentially
  * allowing us to compile the TF-A with -O0 without overlapping with U-Boot;
  * also, U-Boot will be able to reclaim the beginning of SRAM for its MMU
@@ -120,7 +112,9 @@
  */
 #define S32G_BL31_OFF_IN_SRAM		0x00700000
 #define BL31_LIMIT			(S32G_SRAM_BASE + S32G_SRAM_SIZE - 1)
-#endif
+/* U-boot address in SRAM */
+#define S32G_BL33_OFF_IN_SRAM		0x20000
+#define S32G_BL33_IMAGE_BASE		(S32G_SRAM_BASE + S32G_BL33_OFF_IN_SRAM)
 
 #define BL31_BASE			(S32G_SRAM_BASE + S32G_BL31_OFF_IN_SRAM)
 
-- 
2.17.1

